### PR TITLE
CVE-2024-45337

### PR DESCRIPTION
**Pull request by vkamlesh**
The change will fix CVE-2023-45288

Vulnerability details:  https://nvd.nist.gov/vuln/detail/CVE-2024-45337
Fix: golang/go#70779

Thank you for contributing to Velero!

# Please add a summary of your change
As per NVD, Applications and libraries which misuse the ServerConfig.PublicKeyCallback callback may be susceptible to an authorization bypass.Specifically, the SSH protocol allows clients to inquire about whether a public key is acceptable before proving control of the corresponding private key. 

# Does your change fix a particular issue?
Yes. It will fix CVE-2023-45288 vulnerability.

Fixes #(issue)

# Please indicate you've done the following:

- [ x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ x] [Created a changelog file (`make new-changelog`)](https://velero.io/docs/main/code-standards/#adding-a-changelog) or comment `/kind changelog-not-required` on this PR.
- [x ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by blackpiglet** [comment]
@vkamlesh 
Please run `go mod tidy` to update the related package versions in the go.mod and go.sum files.
Please take this PR #8519 as an example.

**Comment by kaovilai** [comment]
Please signoff all your commits https://github.com/vmware-tanzu/velero/pull/8514/checks?check_run_id=34572716368
```
git rebase HEAD~2 --signoff
```
and force push.

File: go.mod [hunk 1]
```diff
@@ -45,7 +45,7 @@ require (
 	golang.org/x/mod v0.22.0
 	golang.org/x/net v0.31.0
```
Comment (kaovilai) [inline go.mod:46]: https://pkg.go.dev/vuln/GO-2024-3333
If you could bump this to v0.33.0 as well here that would be cool.

**Comment by blackpiglet** [comment]
Please also help to bump the version of `golang.org/x/net` to `v0.33.0` to fix https://github.com/advisories/GHSA-w32m-9786-jp63.

Please check #8526 for reference.

**Review by kaovilai — CHANGES_REQUESTED** [review]
We don't use vendor in this repo. Please remove vendor folder.

We could remove for you but this PR is from your main branch, please let us know if you'd like us to commit to your main branch for you.

**Comment by kaovilai** [comment]
And reminder to signoff ALL your commits in the PR.

https://github.com/vmware-tanzu/velero/pull/8514/checks?check_run_id=34673896265
```
Commit sha: [829d626](https://github.com/vmware-tanzu/velero/pull/8514/commits/829d626390bda91cf081672c4a03ab5d6aa3e0b8), Author: kamlesh VERMA, Committer: kamlesh VERMA; The sign-off is missing.
Commit sha: [0f0a3dd](https://github.com/vmware-tanzu/velero/pull/8514/commits/0f0a3dde3741eb8f8be51506bb881eafd410e387), Author: kamlesh VERMA, Committer: kamlesh VERMA; The sign-off is missing.
```

**Review by kaovilai — COMMENTED** [review]
This PR is containing a bunch of unrelated changes now.
ie. the number of files changed is not just go.mod go.sum.

I'll open another PR if that is ok with you. Or please let me know if you prefer us to modify this branch.

**Comment by kaovilai** [comment]
Opened https://github.com/vmware-tanzu/velero/pull/8541 which covers the CVEs discussed in this PR.

**Comment by vkamlesh** [comment]
> This PR is containing a bunch of unrelated changes now. ie. the number of files changed is not just go.mod go.sum.
> 
> I'll open another PR if that is ok with you. Or please let me know if you prefer us to modify this branch.

Sure no problem. Let me close this PR.

**Pull request closed without merging**